Tests/BuildTests/BuildPlanTests.swift

```diff
@@ -3041,6 +3041,76 @@ final class BuildPlanTests: XCTestCase {
         }
     }
 
+    func testEnablingRegexLiterals() throws {
+        // Skip the test if the compiler doens't support `-enable-regex-literals`.
```

Comment:
A follow-on PR should extend BuildParameters (or BuildPlan?) to support a set of overriding entries for the set of supported Swift and Clang compiler features, strictly for testing purposes.  We're not actually invoking the compiler here and it would be better to be able to do "what if" testing:  "assuming the compiler supports it, what would be the command line" etc.